api/app.py

```diff
@@ -23,10 +23,15 @@
 
 testing = 'TESTING' in os.environ
 db_url = Configuration.database_url(testing)
-SessionManager.initialize(db_url)
+# Initialize a new database session unless we were told not to
+# (e.g. because a script already initialized it).
+autoinitialize = os.environ.get('AUTOINITIALIZE') != 'False'
+if autoinitialize:
+    SessionManager.initialize(db_url)
 session_factory = SessionManager.sessionmaker(db_url)
 _db = flask_scoped_session(session_factory, app)
-SessionManager.initialize_data(_db)
+if autoinitialize:
+    SessionManager.initialize_data(_db)
```

Comment:
At some point we took out the AUTOINITIALIZE hack (script tests turn off database initialization code because otherwise there's deadlock) because it didn't make a difference. Something in this branch (or possibly my recent core branch that changes how we get the database URL) jiggled it, and deadlock in script tests came back, so I restored the AUTOINITIALIZE hack.